go/libkb/reset_delete.go

```diff
@@ -4,30 +4,70 @@ import (
 	"errors"
 )
 
-func ResetAccount(m MetaContext, username NormalizedUsername, passphrase string) (err error) {
-	defer m.Trace("ResetAccount", func() error { return err })()
-	return resetOrDeleteAccount(m, username, passphrase, "nuke")
+func EnterResetPipeline(mctx MetaContext, username, email string) (err error) {
+	defer mctx.TraceTimed("EnterResetPipeline", func() error { return err })()
+	// TODO CORE-10466, fully support the `uid` and `self` fields, we should
+	// not attempt to enter the pipeline if the user has an active device. We
+	// should establish a session with a passphrase if at is available.
+	self := mctx.HasAnySession()
+	if self { // clear out other args if we have a session.
+		username = ""
+		email = ""
```

Comment:
This is scary to ignore non-empty arguments

Reply:
What do you think about 3 separate functions here and in avdl `EnterResetPipeline{Self,Username,Email}`?

Reply:
let's talk over the flow in person